slsDetectorSoftware/include/Detector.h

```diff
@@ -980,9 +980,9 @@ class Detector {
     void setInjectChannel(const int offsetChannel, const int incrementChannel,
                           Positions pos = {});
 
-    /** [Gotthard2] adu values for each channel */
-    Result<std::vector<int>> getVetoPhoton(const int chipIndex,
-                                           Positions pos = {});
+    /** [Gotthard2] gain indices and adu values for each channel */
+    void getVetoPhoton(const int chipIndex, const std::string &fname,
```

Comment:
I wonder if we should look through the API and rename functions that read or write files to make them stand out from the normal set/get?

Reply:
like setVetoPhotonFile? and getVetoPhotonFile? In this case, the file uploaded and the file downloaded are different.. but yes sure.. maybe in a different PR though..

Reply:
Makes sense!